sql/src/main/java/io/crate/operation/scalar/cast/CastFunctionResolver.java

```diff
@@ -88,14 +93,30 @@
     /**
      * resolve the needed conversion function info based on the wanted return data type
      */
-    public static FunctionInfo functionInfo(DataType dataType, DataType returnType) {
+    public static FunctionInfo functionInfo(DataType dataType, DataType returnType, boolean tryCast) {
         String functionName = functionMap.get(returnType);
         if (functionName == null) {
             throw new IllegalArgumentException(
                     String.format(Locale.ENGLISH, "No cast function found for return type %s",
                             returnType.getName()));
         }
+        functionName = tryCast ? TRY_CAST_PREFIX + functionName : functionName;
         return new FunctionInfo(new FunctionIdent(functionName, ImmutableList.of(dataType)), returnType);
     }
 
+    public static boolean supportsExplicitConversion(DataType returnType) {
+        return functionMap.get(returnType) != null;
+    }
+
+    public static Map<DataType, String> tryFunctionsMap() {
+        if (tryFunctionsMap != null) {
+            return tryFunctionsMap;
+        }
+        tryFunctionsMap = new HashMap<>();
+        for (Map.Entry<DataType, String> function : functionMap.entrySet()) {
```

Comment:
return a view instead of creating a copy ... you can use Maps.transformValues for that